skimage/segmentation/boundaries.py

```diff
@@ -1,41 +1,214 @@
-    boundaries[:, 1:] += label_img[:, 1:] != label_img[:, :-1]
+def _find_boundaries_subpixel(label_img):
+    """See ``find_boundaries(..., mode='subpixel')``.
+
+    Notes
+    -----
+    This function puts in an empty row and column between each *actual*
+    row and column of the image, for a corresponding shape of $2s - 1$
+    for every image dimension of size $s$. These "interstitial" rows
+    and columns are filled as ``True`` if they separate two labels in
+    `label_img`, ``False`` otherwise.
+
+    I used ``view_as_windows`` to get the neighborhood of each pixel.
+    Then I check whether there are two labels or more in that
+    neighborhood.
+    """
+    ndim = label_img.ndim
+    max_label = np.iinfo(label_img.dtype).max
+
+    label_img_expanded = np.zeros([(2 * s - 1) for s in label_img.shape],
+                                  label_img.dtype)
+    pixels = [slice(None, None, 2)] * ndim
+    label_img_expanded[pixels] = label_img
+
+    edges = np.ones(label_img_expanded.shape, dtype=bool)
+    edges[pixels] = False
+    label_img_expanded[edges] = max_label
+    windows = view_as_windows(pad(label_img_expanded, 1,
+                                  mode='constant', constant_values=0),
+                              (3,) * ndim)
+
+    boundaries = np.zeros_like(edges)
+    for index in np.ndindex(label_img_expanded.shape):
+        if edges[index]:
+            values = np.unique(windows[index].ravel())
+            if len(values) > 2: # single value and max_label
+                boundaries[index] = True
     return boundaries
 
 
+def find_boundaries(label_img, connectivity=1, mode='thick', background=0):
+    """Return bool array where boundaries between labeled regions are True.
+
+    Parameters
+    ----------
+    label_img : array of int
+        An array in which different regions are labeled with different
+        integers.
+    connectivity: int in {1, ..., `label_img.ndim`}, optional
+        A pixel is considered a boundary pixel if any of its neighbors
+        has a different label. `connectivity` controls which pixels are
+        considered neighbors. A connectivity of 1 (default) means
+        pixels sharing an edge (in 2D) or a face (in 3D) will be
+        considered neighbors. A connectivity of `label_img.ndim` means
+        pixels sharing a corner will be considered neighbors.
+    mode: string in {'thick', 'inner', 'outer', 'subpixel'}
+        How to mark the boundaries:
+
+        - thick: any pixel not completely surrounded by pixels of the
+        same label (defined by `connectivity`) is marked as a boundary.
+        This results in boundaries that are 2 pixels thick.
+        - inner: outline the pixels *just inside* of objects, leaving
+        background pixels untouched.
+        - outer: outline pixels in the background around object
+        boundaries. When two objects touch, their boundary is also
+        marked.
+        - subpixel: return a doubled image, with pixels *between* the
```

Comment:
This is cool!  I guess a person could use this to detect the coordinates of the non-zero pixels, scale by 0.5, and plot it in matplotlib.